sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveSqlParser.scala

```diff
@@ -272,10 +272,7 @@ class HiveSqlAstBuilder extends SparkSqlAstBuilder {
       name: String,
       expressions: Seq[Expression],
       ctx: LateralViewContext): Generator = {
-    val info = Option(FunctionRegistry.getFunctionInfo(name.toLowerCase)).getOrElse {
-      throw new ParseException(s"Couldn't find Generator function '$name'", ctx)
-    }
-    HiveGenericUDTF(name, new HiveFunctionWrapper(info.getFunctionClass.getName), expressions)
+    UnresolvedGenerator(name, expressions)
```

Comment:
yeah, I have considered to do this to avoid passing `FunctionRegistry` into `HiveSqlParser`, but not do it yet.